geonode/metadata/api/views.py

```diff
@@ -0,0 +1,272 @@
+                        else "The resource was updated successfully"
+                    ),
+                    "extraErrors": errors,
+                }
+
+                return Response(response, status=400 if errors else 200)
+
+        except ResourceBase.DoesNotExist:
+            result = {"message": "The dataset was not found"}
+            return Response(result, status=404)
+
+
+def tkeywords_autocomplete(request: WSGIRequest, thesaurusid):
+
+    lang = remove_country_from_languagecode(get_language())
+    all_keywords_qs = ThesaurusKeyword.objects.filter(thesaurus_id=thesaurusid)
+
+    # try find results found for given language e.g. (en-us) if no results found remove country code from language to (en) and try again
+    localized_k_ids_qs = ThesaurusKeywordLabel.objects.filter(lang=lang, keyword_id__in=all_keywords_qs).values(
+        "keyword_id"
+    )
+
+    # consider all the keywords that do not have a translation in the requested language
+    keywords_not_translated_qs = (
+        all_keywords_qs.exclude(id__in=localized_k_ids_qs).order_by("id").distinct("id").values("id")
+    )
+
+    qs = ThesaurusKeywordLabel.objects.filter(lang=lang, keyword_id__in=all_keywords_qs).order_by("label")
+    # if q := request.query_params.get("q", None):
+    if q := request.GET.get("q", None):
+        qs = qs.filter(label__istartswith=q)
+
+    ret = []
+    for tkl in qs.all():
+        ret.append(
+            {
+                "id": tkl.keyword.about,
+                "label": tkl.label,
+            }
+        )
+    for tk in all_keywords_qs.filter(id__in=keywords_not_translated_qs).order_by("alt_label").all():
+        ret.append(
+            {
+                "id": tk.about,
+                "label": f"! {tk.alt_label}",
+            }
+        )
+
+    return JsonResponse({"results": ret})
+
+
+def categories_autocomplete(request: WSGIRequest):
+    qs = TopicCategory.objects.order_by("gn_description")
+
+    if q := request.GET.get("q", None):
+        qs = qs.filter(gn_description__istartswith=q)
+
+    ret = []
+    for record in qs.all():
+        ret.append(
+            {
+                "id": record.identifier,
+                "label": _(record.gn_description),
+            }
+        )
+
+    return JsonResponse({"results": ret})
+
+
+def licenses_autocomplete(request: WSGIRequest):
+    qs = License.objects.order_by("name")
+
+    if q := request.GET.get("q", None):
+        qs = qs.filter(name__istartswith=q)
+
+    ret = []
+    for record in qs.all():
+        ret.append(
+            {
+                "id": record.identifier,
+                "label": _(record.name),
+            }
+        )
+
+    return JsonResponse({"results": ret})
+
+
+class ProfileAutocomplete(autocomplete.Select2QuerySetView):
+    def get_queryset(self):
+        if self.request and self.request.user:
+            qs = get_available_users(self.request.user)
+        else:
+            qs = get_user_model().objects.all()
```

Comment:
I dont think makes much sense. Inside the `get_available_users` we filter the user based on what the user can see, but if we dont give a user, will return all. I guess here is better to return 0 users if the user is not provided. (even because even the Anonymous user will match the first `if` so this else will never be met)